refinery/core/views.py

```diff
@@ -925,37 +908,73 @@ def delete(self, request, uuid):
-             )
-        except DataSet.MultipleObjectsReturned as e:
+            raise Http404
+        except ExtendedGroup.MultipleObjectsReturned as e:
             logger.error(e)
-            return HttpResponseServerError(
-                content="Multiple dataSets returned for this request"
-            )
+            raise APIException("Multiple groups returned for this request.")
+
+    def list(self, request):
+        data_set_uuid = request.query_params.get('dataSetUuid')
+        all_perms_flag = request.query_params.get('allPerms', False)
+
+        data_set = get_data_set_for_view_set(data_set_uuid)
 
         public_group = ExtendedGroup.objects.public_group()
         if not ('read_meta_dataset' in get_perms(public_group, data_set) or
                 request.user.has_perm('core.read_meta_dataset', data_set)):
-            return Response(data_set_uuid, status=status.HTTP_401_UNAUTHORIZED)
+            return Response(data_set_uuid, status=status.HTTP_403_FORBIDDEN)
+
+        if all_perms_flag:
+            # all groups user is member of
+            query_set = ExtendedGroup.objects.all()
 
-        groups_with_perms = get_groups_with_perms(data_set)
+        else:
+            # all groups associated with data set and user is a member of
+            query_set = ExtendedGroup.objects.filter(
+                group_ptr__in=get_groups_with_perms(data_set)
+            )
 
-        serializer = GroupSerializer(groups_with_perms, many=True,
-                                     context={'data_set': data_set})
+        member_groups = [group for group in query_set
+                         if request.user in group.user_set.all() and
+                         not group.is_manager_group()]
 
+        serializer = ExtendedGroupSerializer(member_groups,
+                                             many=True,
+                                             context={'data_set': data_set})
         return Response(serializer.data)
 
+    def partial_update(self, request, uuid, format=None):
+        data_set_uuid = request.data.get('dataSetUuid')
+        group = self.get_object(uuid)
+
+        data_set = get_data_set_for_view_set(data_set_uuid)
+        if data_set.get_owner() != request.user:
+            return Response(data_set_uuid, status=status.HTTP_403_FORBIDDEN)
+
+        group_perm_update = request.data.get('perm_list')
+        # remove all perms
+        data_set.unshare(group)
+        if group_perm_update.get('change'):
+            # fields for share method: read_only, read_meta only
+            data_set.share(group, False, False)
+        elif group_perm_update.get('read'):
+            data_set.share(group, True, False)
+        elif group_perm_update.get('read_meta'):
+            data_set.share(group, False, True)
+
+        serializer = ExtendedGroupSerializer(group,
+                                             context={'data_set': data_set})
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
```

Comment:
Is it necessary to set HTTP 200 on a response explicitly? It looks like that's the default value: https://www.django-rest-framework.org/api-guide/responses/#response